DXRModules/DeusEx/Classes/DXRBingoCampaign.uc

```diff
@@ -0,0 +1,257 @@
+
+function PreFirstEntry()
+{
+    local DXREvents events;
+    local BlackHelicopter jock;
+    local Switch1 button;
+    local PayPhone invisibleWall;
+    local #var(prefix)FlagTrigger ft;
+
+    if (!dxr.flags.IsBingoCampaignMode()) return;
+    Super.PreFirstEntry();
+
+    foreach AllActors(class'DXREvents', events) {
+        switch (dxr.localURL) {
+            case "02_NYC_BATTERYPARK":
+                dxr.flags.settings.starting_map = 20;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "03_NYC_UNATCOISLAND":
+                dxr.flags.settings.starting_map = 30;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "03_NYC_AIRFIELD":
+                foreach AllActors(class'BlackHelicopter', jock) {
+                    AddBingoEventBlocker(jock, 'DXRando_Mission03_BingoCompleted');
+                    break;
+                }
+                break;
+            case "04_NYC_UNATCOISLAND":
+                dxr.flags.settings.starting_map = 40;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "05_NYC_UNATCOMJ12LAB":
+                if (!dxr.flagbase.GetBool('DXRando_Mission04_BingoCompleted')) {
+                    SetTimer(1.0, true);
+                }
+                dxr.flags.settings.starting_map = 50;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "06_HONGKONG_HELIBASE":
+                dxr.flags.settings.starting_map = 60;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "08_NYC_STREET":
+                dxr.flags.settings.starting_map = 80;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "09_NYC_Dockyard":
+                dxr.flags.settings.starting_map = 90;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "10_PARIS_CATACOMBS":
+                dxr.flags.settings.starting_map = 100;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "10_PARIS_CHATEAU":
+                if (class'DXRMapVariants'.static.IsVanillaMaps(player())) {
+                    invisibleWall = PayPhone(Spawnm(class'PayPhone',, 'invisible_wall', vect(-999.7, -4484.5, -174.4)));
+                    invisibleWall.bHighlight = false;
+                    invisibleWall.SetCollisionSize(100.0, 150.0);
+                }
+                break;
+            case "11_PARIS_CATHEDRAL":
+                if (!dxr.flagbase.GetBool('DXRando_Mission10_BingoCompleted')) {
+                    SetTimer(1.0, true);
```

Comment:
Is this necessary if you're blocked from proceeding by the invisible wall?

Reply:
It is, yes. The invisible wall is only spawned on vanilla maps. This is to make the mode compatible with Revision etc.

Reply:
You aren't including the module in Revision though, and the correct solution would be to add the invisible wall in Revision as well in that case

Reply:
I think it's maybe safe to include the module in all versions now. But yes, ideally there would also be a special case for Revision as well.

Reply:
> "explode on the next level if you didn't finish the goal" seems bad

No argument from me, I don't like it either.